returnn/hdf.py

```diff
@@ -346,6 +364,296 @@ def run(self):
+            The job creates one HDF per split.
+        :param ffmpeg_output_args: Optional additional arguments to pass to FFmpeg.
+            These arguments are passed to a second FFmpeg call that can be used to
+            optionally compress the audio data and store it in an encoded form.
+            E.g. to compress the audio data to OGG Vorbis, you can use:
+            `ffmpeg_output_args=("-c:a", "libvorbis", "-qscale:a", "0", "-f", "ogg")`.
+            When these arguments are set, the audio data is always written as raw bytes
+            into the HDF (i.e. `dtype=numpy.uint8`).
+            For this reason, `output_dtype` must be set to `None` when using this argument.
+        :param multi_channel_strategy: defines what should happen to multi-channel audio files.
+            Currently implemented are:
+            Mixdown(): Mix down all channels to one channel, default.
+            PickNth(n): Takes audio from n-th channel.
+        :param output_dtype: dtype that should be written in the hdf (supports float64, float32, int16).
+            If writing compressed data, must be set to None as the compressed audio is always written as uint8 (raw bytes).
+        :param returnn_root: RETURNN repository
+        :param rounding: defines how timestamps should be rounded if they do not exactly fall onto a sample:
+            start_and_duration will round down the start time and the duration of the segment
+            rasr_compatible will round up the start time and round down the end time
+        :param round_factor: do the rounding based on a sampling rate that is scaled down by this factor
+        :param target_sampling_rate: desired sampling rate for the HDF, data will be resampled to this rate if needed
+        :param concurrent: Split up the list of splits into this many concurrent jobs.
+            Recommended is about one unit of concurrency per 1000h of audio.
+            This value affects how I/O efficient the job is. With increasing concurrency
+            the I/O efficiency decreases as recordings may end up having to be read multiple
+            times from disk.
+            Within job concurrency is handled by the multiprocessing library, using `num_workers`
+            as parallelism factor.
+            Note that within-job concurrency is more I/O efficient than between-job concurrency,
+            so prefer increasing `num_workers` over increasing `concurrent`, when possible.
+        :param cpu_rqmt: How many CPUs to assign to the job.
+            The job is mainly I/O limited, so it's okay to assign fewer CPUs than the number of worker processes.
+        :param mem_rqmt: How much memory to assign to the job.
+        :param num_workers: Num subprocs (multiprocessing.Pool size) used for parallel recording processing.
+            It can be increased to e.g. match the number of CPU cores on a big cluster machine,
+            and the job will stay I/O efficient.
```

Comment:
Now you have both `cpu_rqmt` and `num_workers`? Do you really need that? My suggestion was to either have `cpu_rqmt` or to use `num_workers` (but definitely not both). Whatever of that you choose, the code should reflect that exactly. I.e. if you specify `num_workers`, that should be what the pool size will be exactly in the end, but `rqmt["cpu"]` can be derived from `num_workers`. If you specify `cpu_rqmt`, then that would be exactly the `rqmt["cpu"]`, and you can infer `num_workers` somehow from it. Both options would be fine with me. But you somehow mixed everything up (it was neither the num workers nor the cpu rqmt).

Reply:
I exposed both values because I think it is difficult to have a value that properly represents the relation between the number of worker procs and the CPU requirement that is correct for all use cases.

Reply:
But before, you also only had one option.

I think there should be a reasonable default for the other value when you specify one of it. E.g. when you only specify `num_workers`, it should set a reasonable cpu rqmt based on it (and not the hardcoded value you have now).

And then, also I don't think you need `cpu_rqmt` as an option here. For other things, for the rare case that you really want to overwrite this, you can simply do:
```python
job = BlissToAudioHDFJob(...)
job.rqmt["cpu"] = ...
```

@michelwi made a very similar argument [here](https://github.com/rwth-i6/i6_core/pull/599#discussion_r2090792950).

I think now it's just unnecessary complicated (having two options) and also not reasonable that cpu rqmt are independent of num_workers by default.